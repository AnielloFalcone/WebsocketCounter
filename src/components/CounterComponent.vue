<template>
    <div id="counter">
        <div class="counter-container">
            {{counter}}
        </div>
    </div>
</template>

<script>
    export default {
        name: "CounterComponent",
        props: ['counter']
    }
</script>

<style lang="scss" scoped>
    #counter {
        width: 20%;
        font-size: 5em;
        display: flex;
        justify-content: center;

        .counter-container {
            height: max-content;
            position: sticky;
            top: 60px;
        }
    }
</style>
