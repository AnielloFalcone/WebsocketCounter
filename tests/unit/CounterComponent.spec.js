import { shallowMount } from "@vue/test-utils";
import CounterComponent from "../../src/components/CounterComponent";

describe("CounterComponent.vue", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount(CounterComponent, {
            propsData: {
                counter: 6
            },
        });
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it("renders CounterComponent when passed", () => {
        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.element).toBeDefined();
        expect(wrapper.element).toMatchSnapshot();
    });

    it('props should be defined', () => {
        expect(wrapper.props().counter).toBe(6);
    });
});
