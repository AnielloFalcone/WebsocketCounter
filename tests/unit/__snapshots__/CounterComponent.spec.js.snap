// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent.vue renders CounterComponent when passed 1`] = `
<div
  id="counter"
>
  <div
    class="counter-container"
  >
    
        6
    
  </div>
</div>
`;
